lib/Sema/CSApply.cpp

```diff
@@ -272,6 +272,20 @@ void ConstraintSystem::propagateLValueAccessKind(Expr *E,
                                allowOverwrite);
 }
 
+bool ConstraintSystem::isTypeReference(Expr *E) {
+  return E->isTypeReference([&](const Expr &E) -> Type { return getType(&E); });
```

Comment:
I'll take a look at that. For some reason I think I tried that and hit issues, but I could be misremembering.